cmake/kokkos_arch.cmake

```diff
@@ -1,7 +1,7 @@
 
-FUNCTION(KOKKOS_ARCH_OPTION SUFFIX DEV_TYPE DESCRIPTION)
+FUNCTION(KOKKOS_ARCH_OPTION SUFFIX DEV_TYPE DESCRIPTION DEPENDENCY)
```

Comment:
to improve readability (and save on future confusion) would it be worth having this argument optional and use the `cmake_parse_arguments` to make it a keyword argument?

e.g. for host just have ` KOKKOS_ARCH_OPTION(${ARCH} HOST "${LABEL}")`

but for devices dependent on an arch have `KOKKOS_ARCH_OPTION(KEPLER30        GPU  "NVIDIA Kepler generation CC 3.0" DEPENDS_ON "KOKKOS_SHOW_CUDA_ARCHS")`

Reply:
I don't like the `cmake_parse_arguments` syntax but I'm happy to look into it if you think it's worth it.

Reply:
no, I think it's fine